src/mtype.d

```diff
@@ -9497,27 +9453,69 @@ public:
         return params;
     }
 
+    extern (D):
```

Comment:
Because the _foreach function being D does not make the delegate parameter D.

If there's a mismatch in a .h file, and somebody actually uses it, there'll be a linker error. So I'm not concerned about it until/unless that happens. There's far more exposed in the .h files than needs to be, and as I wrote before, that is a subject for a different PR than this one.